### PR TITLE
Organize crates into folders

### PR DESCRIPTION
**Pull request by emilk**
### What
* Closes https://github.com/rerun-io/rerun/issues/5888

This organizes the crates into five folders:
* `crates/build`: build tools
* `crates/top`: top-level crates (`re_sdk`, `rerun`, `rerun_c`, `rerun-cli`)
* `crates/viewer`: everything visual and UI
* `crates/store`: how data is stored, transmitted, indexed, plus type definitions
* `crates/utils`: bottom-level crates, perhaps useful to others

This is a very rough split, and better ones are certainly possible, but I think this is at least a good start.

Future work: lint on wether or not crates in one folder may depend on crates in another.

```
 crates
 ├─ build
 │  ├─ re_build_info
 │  ├─ re_build_tools
 │  ├─ re_dev_tools
 │  └─ re_types_builder
 ├─ store
 │  ├─ re_chunk
 │  ├─ re_chunk_store
 │  ├─ re_data_loader
 │  ├─ re_data_source
 │  ├─ re_entity_db
 │  ├─ re_format_arrow
 │  ├─ re_log_encoding
 │  ├─ re_log_types
 │  ├─ re_query
 │  ├─ re_sdk_comms
 │  ├─ re_types
 │  ├─ re_types_blueprint
 │  ├─ re_types_core
 │  └─ re_ws_comms
 ├─ top
 │  ├─ re_sdk
 │  ├─ rerun
 │  ├─ rerun-cli
 │  └─ rerun_c
 ├─ utils
 │  ├─ re_analytics
 │  ├─ re_case
 │  ├─ re_crash_handler
 │  ├─ re_error
 │  ├─ re_format
 │  ├─ re_int_histogram
 │  ├─ re_log
 │  ├─ re_memory
 │  ├─ re_smart_channel
 │  ├─ re_string_interner
 │  ├─ re_tracing
 │  └─ re_tuid
 └─ viewer
    ├─ re_blueprint_tree
    ├─ re_context_menu
    ├─ re_data_ui
    ├─ re_edit_ui
    ├─ re_renderer
    ├─ re_renderer_examples
    ├─ re_selection_panel
    ├─ re_space_view
    ├─ re_space_view_bar_chart
    ├─ re_space_view_dataframe
    ├─ re_space_view_spatial
    ├─ re_space_view_tensor
    ├─ re_space_view_text_document
    ├─ re_space_view_text_log
    ├─ re_space_view_time_series
    ├─ re_time_panel
    ├─ re_ui
    ├─ re_viewer
    ├─ re_viewer_context
    ├─ re_viewport
    ├─ re_viewport_blueprint
    └─ re_web_viewer_server
```

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6821?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6821?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6821)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

**Comment by emilk** [comment]
@rerun-bot full-check

**Comment by emilk** [comment]
@rerun-bot full-check

**Comment by emilk** [comment]
@rerun-bot full-check

File: crates/build/re_types_builder/src/bin/build_re_types.rs [hunk 1]
```diff
@@ -72,8 +72,14 @@ fn main() {
     let workspace_dir = Utf8Path::new(env!("CARGO_MANIFEST_DIR"))
         .parent()
         .and_then(|p| p.parent())
+        .and_then(|p| p.parent())
         .unwrap();
 
+    assert!(
+        workspace_dir.join("CODE_OF_CONDUCT.md").exists(),
+        "ailed to find workspace root"
```
Comment (Wumpf) [inline crates/build/re_types_builder/src/bin/build_re_types.rs:80]:
```suggestion
        "failed to find workspace root"
```

**Review by Wumpf — APPROVED** [review]
lgtm and I think this is a good split, but we should wait for:
* more opinions from the team on the split
* check if someone has something big in flight that will be harder to merge/rebase than this

**Comment by emilk** [comment]
> check if someone has something big in flight that will be harder to merge/rebase than this

Rebasing on moved files is quite painless - I know, I did it three times while working on this PR :)

**Pull request merged**